plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -491,6 +520,44 @@ protected List<CastTestCase> supportedCastTypePushdown()
                 new CastTestCase("c_decimal_negative", "integer", Optional.of("c_integer")),
 
+                new CastTestCase("c_char_10", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_char_10", "char(256)", Optional.of("c_bpchar")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nvarchar_10", "varchar(50)", Optional.of("c_nvarchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_text")),
+
+                new CastTestCase("c_char_50", "char(10)", Optional.of("c_char_10")),
+                new CastTestCase("c_bpchar", "char(10)", Optional.of("c_char_10")),
+                new CastTestCase("c_varchar_50", "varchar(10)", Optional.of("c_varchar_10")),
+                new CastTestCase("c_nvarchar_50", "varchar(10)", Optional.of("c_nvarchar_10")),
+                new CastTestCase("c_text", "varchar(10)", Optional.of("c_varchar_10")),
+
+                new CastTestCase("c_char", "varchar(10)", Optional.of("c_varchar_10")),
+                new CastTestCase("c_char", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_char", "varchar(65535)", Optional.of("c_varchar_65535")),
+                new CastTestCase("c_char", "varchar(65535)", Optional.of("c_varchar_65535")),
+
+                new CastTestCase("c_char_10", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_char_10", "char(256)", Optional.of("c_bpchar")),
+                new CastTestCase("c_char", "char(4096)", Optional.of("c_char_4096")),
+                new CastTestCase("c_char", "char(1)", Optional.of("c_nchar")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nvarchar_10", "varchar(50)", Optional.of("c_nvarchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_text")),
+
+                new CastTestCase("c_varchar_50", "varchar(10)", Optional.of("c_varchar_10")),
+
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(65535)", Optional.of("c_varchar_65535")),
+                new CastTestCase("c_varchar_10", "varchar(256)", Optional.of("c_nvarchar")),
+                new CastTestCase("c_varchar_10", "varchar(10)", Optional.of("c_nvarchar_10")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_nvarchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(65535)", Optional.of("c_nvarchar_65535")),
+                new CastTestCase("c_varchar_10", "varchar(256)", Optional.of("c_text")),
+
+                new CastTestCase("c_varchar_unicode", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nvarchar_unicode", "varchar(50)", Optional.of("c_varchar_50")),
+
```

Comment:
can we add test case for cast to `varchar` (unbounded varchar)?

Reply:
Something like https://github.com/mayankvadariya/trino/blob/41ba30e202dd5ab464d3d6bef7172e518d286db0/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleCastPushdown.java#L411-L445